chain/yoda/context.go

```diff
@@ -47,6 +47,10 @@ type Context struct {
 
 	dataSourceCache *sync.Map
 	pendingRequests map[types.RequestID]bool
+
+	handlingGauge  int64
```

Comment:
I'm not sure if these variables need to be threadsafe (using `sync/atomic`)? `handleRawRequest` for examples can be called concurrently by multiple goroutines.